### ISSUE TITLE
Authentication - Authorize Method parameters

### ISSUE THREAD

**Issue by lmborione** [autodesk-platform-services/aps-sdk-node#75]
Dear all,

The authorize method used in the Authentication client ```Authentication/source/custom-code/AuthenticationClient.ts``` : 
```
    public authorize(clientId: string, responseType: ResponseType, redirectUri: string, scopes: Array<Scopes>, optionalArgs?: { state?: string, nonce?: string, responseMode?: string, prompt?: string, authoptions?: string, codeChallenge?: string, codeChallengeMethod?: string }): string {
        const url = this.tokenApi.authorize(clientId, responseType, redirectUri, optionalArgs?.state, optionalArgs?.nonce, scopes, optionalArgs?.responseMode, optionalArgs?.prompt, optionalArgs?.authoptions, optionalArgs?.codeChallenge, optionalArgs?.codeChallengeMethod);
        return url;
    }
```
Does not follow the method signature in the Token Api ```Authentication/source/api/token-api.ts``` :
```
public authorize(clientId: string, responseType: ResponseType, redirectUri: string, nonce?: string, state?: string, scopes?: Array<Scopes>, responseMode?: string, prompt?: string, authoptions?: string, codeChallenge?: string, codeChallengeMethod?: string) {
      this.logger.logInfo("Entered into authorize ");
      try {
        const url =  TokenApiFp(this.sdkManager).authorize(clientId, responseType, redirectUri, nonce, state, scopes, responseMode, prompt, authoptions, codeChallenge, codeChallengeMethod);
        this.logger.logInfo(`authorize Request completed successfully.`);
        return url;
      } catch (error) {
        if (error.response) {
            const errorMessage = JSON.stringify(error.response.data);
            this.logger.logError(`authorize Request failed with status : ${error.response.status} and statusText : ${error.response.statusText} and error message: ${errorMessage}`);
            throw new AuthenticationApiError(`authorize Request failed with status : ${error.response.status} and error message: ${errorMessage}`, error);
        } else if (error.request) {
            this.logger.logError(`authorize Request failed with no response received: ${error.request}`);
            throw new AuthenticationApiError(`authorize Request failed with no response received: ${error.request}`, error);
        }
        throw error;
      }
    }
```

4th arguments should be ```optionalArgs?.nonce``` and 5th ```optionalArgs?.state```

Thanks

**Comment by sajith-subramanian** [comment]
Thanks for reporting this @lmborione. WIll log this one for an immediate fix.

**Comment by sajith-subramanian** [comment]
Fixed in the latest version v1.0.0
